### PR TITLE
feat(awful: spawn): add 'easy_async' wrapper

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

File: lib/awful/spawn.lua [hunk 1]
```diff
@@ -139,6 +139,58 @@ function spawn.with_line_callback(cmd, callbacks)
     return pid
 end
 
+--- Spawn a program and asynchronously and capture its output
```
Comment (psychon) [inline lib/awful/spawn.lua]: I'm not really good at English, but shouldn't it be: `Asynchronously spawn a program and capture its output`?

**Comment by psychon** [comment]
Fine by me: :+1:

**Comment by psychon** [comment]
@blueyed Should look at the docs for this one. Perhaps he can come up with a nicer explanation of the callback arguments. I added this to the "next" milestone for now.

**Comment by actionless** [comment]
@blueyed so what about the docstrings?

**Comment by psychon** [comment]
I merged this as-is. If there is a better way to document this, we can change that later.

**Comment by blueyed** [comment]
I think the documentation is fine: http://awesomewm.github.io/apidoc/libraries/awful.spawn.html#easy_async.

**Comment by psychon** [comment]
Wow, that looks surprisingly good and definitely not what I expected. Sorry for my confusion.

**Pull request merged**